app/src/test/java/org/oppia/android/app/player/state/StateFragmentLocalTest.kt

```diff
@@ -1048,6 +1051,52 @@ class StateFragmentLocalTest {
     }
   }
 
+  @Test
+  @Config(qualifiers = "+port")
+  fun testStateFragment_finishExploration_changePortToLand_endOfSessionConfettiIsDisplayedAgain() {
+    launchForExploration(FRACTIONS_EXPLORATION_ID_1).use {
+      startPlayingExploration()
+      playThroughAllStates()
+      clickContinueButton()
+      onView(withId(R.id.full_screen_confetti_view)).check(
+        matches(
+          hasExpectedNumberOfActiveSystems(numSystems = 2)
+        )
+      )
+
+      onView(isRoot()).perform(orientationLandscape())
```

Comment:
Suggest temporarily copying in the utility from #2572 temporarily into your branch & using that here & below just to make sure that this works correctly in Robolectric with a correct rotation.

Reply:
Ah will do -- I'm getting errors from the rotation now, but I think the might be coming from the actual rotating.. working to debug this

Reply:
It's unclear whether the error is from the confetti tests or the rotation utility so I'll hold off on adding those changes so this isn't blocked